v-next/hardhat-ethers/src/internal/hardhat-helpers/hardhat-helpers.ts

```diff
@@ -0,0 +1,473 @@
+import type {
```

Comment:
As mentioned in the PR description:

in the old V2 code the hardhat helpers were separated in functions, [see here](https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-ethers/src/internal/helpers.ts). In V3 I grouped them into a class, to simplify the usage of the dependencies that before were imported from the HardhatRuntimeEnvironment. For example, before the functionalities available in the `hre` (like hre.artifacts ) had to be passed several times across functions, now they can be directly accessed via `this`